adapters/repos/db/index_queue_test.go

```diff
@@ -304,6 +306,54 @@ func TestIndexQueue(t *testing.T) {
 
 		require.EqualValues(t, []uint64{1, 2, 3}, idx.IDs())
 	})
+
+	t.Run("updates the shard state", func(t *testing.T) {
+		var idx mockBatchIndexer
+		indexed := make(chan struct{})
+		idx.addBatchFn = func(id []uint64, vector [][]float32) error {
+			close(indexed)
+			return nil
+		}
+
+		updated := make(chan string)
+		shard := mockShard{
+			compareAndSwapStatusFn: func(old, new string) (storagestate.Status, error) {
+				updated <- new
+				return storagestate.Status(new), nil
+			},
+		}
+
+		q, err := NewIndexQueue(&shard, &idx, startWorker(t), IndexQueueOptions{
+			BatchSize:     2,
+			IndexInterval: 10 * time.Millisecond,
+		})
+		require.NoError(t, err)
+		defer q.Close()
+
+		for i := uint64(0); i < 2; i++ {
+			pushVector(t, ctx, q, i+1, 1, 2, 3)
+		}
+
+		select {
+		case newState := <-updated:
+			require.Equal(t, storagestate.StatusIndexing.String(), newState)
+		case <-time.After(100 * time.Millisecond):
+			t.Fatal("shard state should have been updated after 10ms")
```

Comment:
minor: error message duration doesn't match threshold 100ms vs 10ms

Reply:
Actually it's 10ms because the IndexInterval is 10ms, but I wait 100ms just to be sure we don't have a flaky test (cpu throttling on the CI, etc.)